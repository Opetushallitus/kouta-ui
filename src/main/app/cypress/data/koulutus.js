import { merge } from 'lodash';

const getBaseFields = () => {
  return {
    oid: '1.2.246.562.13.00000000000000000072',
    koulutustyyppi: 'amm',
    koulutusKoodiUri: 'koulutus_0#1',
    tila: 'tallennettu',
    tarjoajat: ['1.2.246.562.10.81934895871'],
    nimi: {
      en: 'Vocational qualification in Agriculture',
      fi: 'Maatalousalan perustutkinto',
      sv: 'Grundexamen inom lantbruksbranschen',
    },
    metadata: {
      tyyppi: 'amm',
      kuvaus: {},
      lisatiedot: [
        {
          otsikkoKoodiUri: 'koulutuksenlisatiedot_0#1',
          teksti: {
            fi: 'koulutuksenlisatiedot_0 kuvaus',
          },
        },
      ],
    },
    julkinen: true,
    muokkaaja: '1.2.246.562.24.62301161440',
    organisaatioOid: '1.2.246.562.10.594252633210',
    kielivalinta: ['fi', 'sv'],
    modified: '2019-04-01T13:01',
  };
};

const getAmmatillinenFields = ({ tyyppi }) => {
  return merge(getBaseFields(), {
    koulutustyyppi: tyyppi,
    koulutusKoodiUri: 'koulutus_0#1',
    tila: 'tallennettu',
    nimi: {
      en: 'Vocational qualification in Agriculture',
      fi: 'Maatalousalan perustutkinto',
      sv: 'Grundexamen inom lantbruksbranschen',
    },
    metadata: { tyyppi },
  });
};

const getKorkeakouluFields = ({ tyyppi }) => {
  return merge(getBaseFields(), {
    johtaaTutkintoon: true,
    koulutustyyppi: tyyppi,
    koulutusKoodiUri: 'koulutus_0#1',
    nimi: { fi: 'Fi nimi', sv: 'Sv nimi' },
    metadata: {
      tyyppi,
      kuvaus: { fi: 'Fi kuvaus', sv: 'Sv kuvaus' },
      tutkintonimikeKoodiUrit: ['tutkintonimikekk_1#1', 'tutkintonimikekk_2#1'],
      opintojenLaajuusKoodiUri: 'opintojenlaajuus_1#1',
      kuvauksenNimi: { fi: 'Fi kuvauksen nimi', sv: 'Sv kuvauksen nimi' },
      koulutusalaKoodiUrit: [
        'kansallinenkoulutusluokitus2016koulutusalataso2_1#1',
      ],
    },
  });
};

const getLukioFields = ({ tyyppi }) => getAmmatillinenFields({ tyyppi });

export default ({ tyyppi = 'amm' } = {}) => {
  if (tyyppi === 'amm') {
    return getAmmatillinenFields({ tyyppi });
  } else if (['yo', 'amk'].includes(tyyppi)) {
    return getKorkeakouluFields({ tyyppi });
  } else if (tyyppi === 'lk') {
    return getLukioFields({ tyyppi });
  }

  return getBaseFields();
};
