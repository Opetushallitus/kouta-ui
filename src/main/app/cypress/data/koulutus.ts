import { merge } from 'lodash/fp';

const getBaseFields = ({ tyyppi = 'amm' } = {}) => {
  return {
    oid: '1.2.246.562.13.00000000000000000072',
    koulutustyyppi: tyyppi,
    koulutuksetKoodiUri: ['koulutus_0#1'],
    tila: 'tallennettu',
    tarjoajat: ['1.2.246.562.10.81934895871'],
    nimi: {
      en: 'Vocational qualification in Agriculture',
      fi: 'Maatalousalan perustutkinto',
      sv: 'Grundexamen inom lantbruksbranschen',
    },
    metadata: {
      tyyppi: tyyppi,
      kuvaus: {},
      lisatiedot: [
        {
          otsikkoKoodiUri: 'koulutuksenlisatiedot_0#1',
          teksti: {
            fi: 'koulutuksenlisatiedot_0 kuvaus',
          },
        },
      ],
    },
    julkinen: true,
    muokkaaja: '1.2.246.562.24.62301161440',
    organisaatioOid: '1.2.246.562.10.594252633210',
    kielivalinta: ['fi', 'sv'],
    modified: '2019-04-01T13:01',
  };
};

const getAmmatillinenFields = ({ tyyppi = 'amm' }) => {
  return merge(getBaseFields(), {
    oid: '1.2.246.562.13.00000000000000000599',
    johtaaTutkintoon: false,
    koulutustyyppi: tyyppi,
    koulutuksetKoodiUri: ['koulutus_351107#12'],
    tila: 'tallennettu',
    tarjoajat: [],
    nimi: {
      en: 'Vocational qualification in Mining',
      fi: 'Kaivosalan perustutkinto',
      sv: 'Grundexamen inom gruvbranschen',
    },
    metadata: {
      tyyppi: tyyppi,
      kuvaus: {},
      lisatiedot: [],
    },
    julkinen: true,
    esikatselu: true,
    muokkaaja: '1.2.246.562.24.26603962037',
    organisaatioOid: '1.2.246.562.10.60198812368',
    kielivalinta: ['fi', 'sv'],
    ePerusteId: 6777660,
    modified: '2020-09-21T16:27',
  });
};

const getKorkeakouluFields = ({ tyyppi }) => {
  return merge(getBaseFields(), {
    johtaaTutkintoon: true,
    koulutustyyppi: tyyppi,
    koulutuksetKoodiUri: ['koulutus_671112#12'],
    nimi: { fi: 'Fysioterapeutti (AMK)', sv: 'Fysioterapeut (YH)' },
    metadata: {
      tyyppi,
      kuvaus: { fi: 'Fi kuvaus', sv: 'Sv kuvaus' },
      tutkintonimikeKoodiUrit: ['tutkintonimikekk_1#1', 'tutkintonimikekk_2#1'],
      opintojenLaajuusKoodiUri: 'opintojenlaajuus_1#1',
      kuvauksenNimi: { fi: 'Fi kuvauksen nimi', sv: 'Sv kuvauksen nimi' },
      koulutusalaKoodiUrit: [
        'kansallinenkoulutusluokitus2016koulutusalataso2_052#1',
      ],
    },
  });
};

const getLukioFields = ({ tyyppi }) => {
  return merge(getBaseFields({ tyyppi }), {
    koulutuksetKoodiUri: ['koulutus_309902#7'],
    nimi: { fi: 'Lukion oppimäärä' },
    metadata: {
      opintojenLaajuusKoodiUri: 'opintojenlaajuus_150',
    },
  });
};

const getTuvaFields = ({ tyyppi }) => {
  return merge(getBaseFields({ tyyppi }), {
    nimi: { fi: 'Tutkintokoulutukseen valmentava koulutus (TUVA)' },
    metadata: {
      opintojenLaajuusKoodiUri: 'opintojenlaajuus_38',
      kuvaus: { fi: 'kuvausteksti' },
    },
  });
};

const getTelmaFields = ({ tyyppi }) => {
  return merge(getBaseFields({ tyyppi }), {
    nimi: { fi: 'Työhön ja itsenäiseen elämään valmentava koulutus (TELMA)' },
    metadata: {
      opintojenLaajuusKoodiUri: 'opintojenlaajuus_60',
      kuvaus: { fi: 'kuvausteksti' },
    },
  });
};

export default ({ tyyppi = 'amm' } = {}) => {
  if (tyyppi.startsWith('amm')) {
    return getAmmatillinenFields({ tyyppi });
  } else if (['yo', 'amk'].includes(tyyppi)) {
    return getKorkeakouluFields({ tyyppi });
  } else if (tyyppi === 'lk') {
    return getLukioFields({ tyyppi });
  } else if (tyyppi === 'tuva') {
    return getTuvaFields({ tyyppi });
  } else if (tyyppi === 'telma') {
    return getTelmaFields({ tyyppi });
  }

  return getBaseFields();
};
