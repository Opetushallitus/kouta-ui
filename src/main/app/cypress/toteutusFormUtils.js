import merge from 'lodash/merge';

import {
  stubKoodistoRoute,
  stubOppijanumerorekisteriHenkiloRoute,
} from './utils';
import organisaatio from './data/organisaatio';
import organisaatioHierarkia from './data/organisaatioHierarkia';
import koodisto from './data/koodisto';

export const stubToteutusFormRoutes = ({ cy, organisaatioOid, perusteId }) => {
  cy.server();

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/hierarkia/hae?oid=${organisaatioOid}**`,
    response: organisaatioHierarkia({ rootOid: organisaatioOid }),
  });

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    response: merge(organisaatio(), {
      oid: organisaatioOid,
    }),
  });

  cy.route({
    method: 'POST',
    url: '**/organisaatio-service/rest/organisaatio/v4/findbyoids',
    response: [
      merge(organisaatio(), {
        oid: organisaatioOid,
      }),
    ],
  });

  stubKoodistoRoute({ koodisto: 'koulutuksenlisatiedot', cy });
  stubKoodistoRoute({ koodisto: 'oppilaitoksenopetuskieli', cy });
  stubKoodistoRoute({ koodisto: 'opetusaikakk', cy });
  stubKoodistoRoute({ koodisto: 'opetuspaikkakk', cy });
  stubKoodistoRoute({ koodisto: 'kausi', cy });
  stubKoodistoRoute({ koodisto: 'kieli', cy });
  stubKoodistoRoute({ koodisto: 'lukiodiplomit', cy });
  stubKoodistoRoute({ koodisto: 'lukiolinjat', cy });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-ylakoodit/koulutustyyppi_*',
    response: koodisto({ koodisto: 'koulutus' }),
  });

  cy.route({
    method: 'GET',
    url: '**/toteutus/list**',
    response: [],
  });

  cy.route({
    method: 'GET',
    url: '**/koodisto-service/rest/codeelement/koulutus_0',
    response: [
      {
        koodiUri: 'koulutus_0',
        versio: 1,
        metadata: [{ kieli: 'fi', nimi: 'Nimi' }],
      },
    ],
  });

  cy.route({
    method: 'GET',
    url:
      '**/eperusteet-service/api/perusteet?tuleva=true&siirtyma=false&voimassaolo=true&poistunut=false&kieli=fi&koulutuskoodi=koulutus_0',
    response: {
      data: [
        {
          kuvaus: { fi: 'koulutus_0 kuvaus' },
          osaamisalat: [
            { uri: 'osaamisala_0', nimi: { fi: 'osaamisala_0 nimi' } },
          ],
          tutkintonimikeKoodiUri: 'nimike_1#1',
          id: perusteId,
        },
      ],
    },
  });

  cy.route({
    method: 'GET',
    url: `**/eperusteet-service/api/perusteet/${perusteId}/osaamisalakuvaukset`,
    response: {
      data: {
        reformi: {},
      },
    },
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-alakoodit/koulutus_0**',
    response: [
      {
        koodiUri: 'okmohjauksenala_1',
        metadata: [{ kieli: 'fi', nimi: 'Ala' }],
      },
      {
        koodiUri: 'opintojenlaajuus_1',
        metadata: [{ kieli: 'fi', nimi: '180' }],
      },
      {
        koodiUri: 'opintojenlaajuusyksikko_1',
        metadata: [{ kieli: 'fi', nimi: 'op' }],
      },
    ],
  });

  cy.route({
    method: 'GET',
    url: '**/ammattinimike/search/**',
    response: [],
  });

  cy.route({
    method: 'GET',
    url: '**/asiasana/search/**',
    response: [],
  });

  stubOppijanumerorekisteriHenkiloRoute({ cy });
};
