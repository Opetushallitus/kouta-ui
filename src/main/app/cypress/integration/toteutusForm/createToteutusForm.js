import merge from 'lodash/merge';

import {
  getRadio,
  getSelectOption,
  getCheckbox,
  chooseKieliversiotLanguages,
  selectOption,
  fillAsyncSelect,
  fillTreeSelect,
  fillDatePickerInput,
} from '../../utils';

import koulutus from '../../data/koulutus';
import { stubToteutusFormRoutes } from '../../toteutusFormUtils';

const jatka = () => {
  cy.getByTestId('jatkaButton').click({ force: true });
};

const fillTilaSection = (tila = 'julkaistu') => {
  cy.getByTestId('tilaSection').within(() => {
    getRadio(tila, cy).check({ force: true });
  });
};

const fillPohjaSection = () => {
  cy.getByTestId('pohjaSection').within(() => {
    jatka();
  });
};

const fillKieliversiotSection = () => {
  cy.getByTestId('kieliversiotSection').within(() => {
    chooseKieliversiotLanguages(['fi'], cy);
    jatka();
  });
};

const fillOpetuskieli = () => {
  cy.getByTestId('opetuskieli').within(() => {
    getCheckbox('oppilaitoksenopetuskieli_0#1', cy).click({ force: true });
    cy.get('textarea').type('opetuskieli kuvaus', { force: true });
  });
};

const fillOpetusaika = () => {
  cy.getByTestId('opetusaika').within(() => {
    getCheckbox('opetusaikakk_0#1', cy).click({ force: true });
    cy.get('textarea').type('opetusaika kuvaus', { force: true });
  });
};

const fillOpetustapa = () => {
  cy.getByTestId('opetustapa').within(() => {
    getCheckbox('opetuspaikkakk_0#1', cy).click({ force: true });
    cy.get('textarea').type('opetustapa kuvaus', { force: true });
  });
};

const fillMaksullisuus = tyyppi => {
  cy.getByTestId('maksullisuus').within(() => {
    getRadio(tyyppi, cy).click({ force: true });
    cy.getByTestId('maksu')
      .find('input')
      .type('10');

    cy.getByTestId('maksullisuusKuvaus').within(() => {
      cy.get('textarea').type('maksullisuus kuvaus', { force: true });
    });
  });
};

const fillStipendi = () => {
  cy.getByTestId('stipendi').within(() => {
    getRadio('kylla', cy).click({ force: true });
    cy.getByTestId('stipendinMaara')
      .find('input')
      .type('20');
    cy.get('textarea').type('stipendi kuvaus', { force: true });
  });
};

const fillKausi = () => {
  cy.getByTestId('koulutuksenAlkamispaivamaara').within(() => {
    fillDatePickerInput('1.1.2019');
  });

  cy.getByTestId('koulutuksenPaattymispaivamaara').within(() => {
    fillDatePickerInput('15.2.2019');
  });
};

const fillOsiot = () => {
  cy.getByTestId('osiotSelect').click();

  cy.getByTestId('osiotSelect').within(() => {
    getSelectOption('koulutuksenlisatiedot_0', cy).click({
      force: true,
    });
  });

  cy.getByTestId('osioKuvaus.koulutuksenlisatiedot_0#1')
    .find('textarea')
    .type('koulutuksenlisatiedot_0 kuvaus', { force: true });
};

const fillCommonJarjestamistiedot = ({ maksullisuusTyyppi = 'kylla' } = {}) => {
  fillOpetuskieli();
  fillOpetusaika();
  fillOpetustapa();
  fillMaksullisuus(maksullisuusTyyppi);
  fillKausi();
  fillOsiot();
};

const fillNayttamistiedotSection = () => {
  cy.getByTestId('nayttamistiedotSection').within(() => {
    cy.getByTestId('ammattinimikkeetSelect').within(() => {
      fillAsyncSelect('ammattinimike', 'Luo kohde');
    });

    cy.getByTestId('avainsanatSelect').within(() => {
      fillAsyncSelect('avainsana', 'Luo kohde');
    });

    jatka();
  });
};

const tallenna = () => {
  cy.getByTestId('tallennaToteutusButton').click({ force: true });
};

const fillJarjestajatSection = () => {
  cy.getByTestId('jarjestamispaikatSection').within(() => {
    cy.getByTestId('jarjestamispaikatSelection').within(() => {
      fillTreeSelect(['4.1.1.1.1.1'], cy);
    });

    jatka();
  });
};

const fillTiedotSection = () => {
  cy.getByTestId('tiedotSection').within(() => {
    cy.getByTestId('toteutuksenNimi')
      .find('input')
      .clear({ force: true })
      .type('toteutuksen nimi', { force: true });

    cy.getByTestId('toteutuksenKuvaus')
      .find('textarea')
      .type('Toteutuksen kuvaus', { force: true });

    jatka();
  });
};

const fillYhteystiedotSection = () => {
  cy.getByTestId('yhteystiedotSection').within(() => {
    cy.getByTestId('lisaaYhteyshenkiloButton').click({ force: true });

    cy.getByTestId('nimi')
      .find('input')
      .type('nimi', { force: true });
    cy.getByTestId('titteli')
      .find('input')
      .type('titteli', { force: true });
    cy.getByTestId('sahkoposti')
      .find('input')
      .type('sÃ¤hkoposti', { force: true });
    cy.getByTestId('puhelinnumero')
      .find('input')
      .type('puhelin', { force: true });
    cy.getByTestId('verkkosivu')
      .find('input')
      .type('verkkosivu', { force: true });

    jatka();
  });
};

const fillKkOsaamisalat = () => {
  cy.getByTestId('lisaaOsaamisalaButton').click({ force: true });
  cy.getByTestId('osaamisalanNimi')
    .find('input')
    .type('osaamisalan nimi', { force: true });
  cy.getByTestId('osaamisalanKuvaus')
    .find('textarea')
    .type('osaamisalan kuvaus', { force: true });
  cy.getByTestId('osaamisalanLinkki')
    .find('input')
    .type('linkki', { force: true });
  cy.getByTestId('osaamisalanOtsikko')
    .find('input')
    .type('osaamisalan otsikko', { force: true });
};

const fillLukiolinjatSection = () => {
  cy.getByTestId('lukiolinjatSection').within(() => {
    cy.getByTestId('lukiolinja').within(() => {
      selectOption('lukiolinjat_0', cy);
    });

    jatka();
  });
};

const fillDiplomi = () => {
  cy.getByTestId('diplomiTyypit').within(() => {
    selectOption('lukiodiplomit_0', cy);
  });

  cy.getByTestId('diplomiKuvaus')
    .find('textarea')
    .type('Diplomi kuvaus', { force: true });
};

const fillKielivalikoima = () => {
  cy.getByTestId('A1A2Kielet').within(() => {
    selectOption('kieli_0', cy);
  });

  cy.getByTestId('B2Kielet').within(() => {
    selectOption('kieli_1', cy);
  });

  cy.getByTestId('aidinkielet').within(() => {
    selectOption('kieli_2', cy);
  });

  cy.getByTestId('B1Kielet').within(() => {
    selectOption('kieli_3', cy);
  });

  cy.getByTestId('B3Kielet').within(() => {
    selectOption('kieli_4', cy);
  });

  cy.getByTestId('muutKielet').within(() => {
    selectOption('kieli_4', cy);
  });
};

describe('createToteutusForm', () => {
  const organisaatioOid = '1.1.1.1.1.1';
  const koulutusOid = '1.2.1.1.1.1';
  const perusteId = '1';

  const testKoulutusFields = {
    oid: koulutusOid,
    organisaatioOid: organisaatioOid,
    koulutusKoodiUri: 'koulutus_0#1',
    tarjoajat: ['4.1.1.1.1.1', '2.1.1.1.1.1'],
  };

  beforeEach(() => {
    stubToteutusFormRoutes({ cy, perusteId, organisaatioOid });

    cy.visit(
      `/organisaatio/${organisaatioOid}/koulutus/${koulutusOid}/toteutus`,
    );
  });

  it('should be able to create ammatillinen toteutus', () => {
    cy.route({
      method: 'GET',
      url: `**/koulutus/${koulutusOid}`,
      response: merge(koulutus({ tyyppi: 'amm' }), testKoulutusFields),
    });

    cy.route({
      method: 'PUT',
      url: '**/toteutus',
      response: {
        oid: '1.2.3.4.5.6',
      },
    }).as('createAmmToteutusResponse');

    fillPohjaSection();
    fillKieliversiotSection();
    fillTiedotSection();

    cy.getByTestId('osaamisalatSection').within(() => {
      cy.getByTestId('osaamisalaSelection').within(() => {
        getCheckbox('osaamisala_0', cy).click({ force: true });
      });

      cy.getByTestId('osaamisalaToggle.osaamisala_0').click({ force: true });

      cy.getByTestId('osaamisalaLinkki.osaamisala_0')
        .find('input')
        .type('osaamisala_0 linkki', { force: true });

      cy.getByTestId('osaamisalaOtsikko.osaamisala_0')
        .find('input')
        .type('osaamisala_0 otsikko', { force: true });

      jatka();
    });

    cy.getByTestId('jarjestamistiedotSection').within(() => {
      fillCommonJarjestamistiedot();
      jatka();
    });

    fillNayttamistiedotSection();
    fillJarjestajatSection();
    fillYhteystiedotSection();
    fillTilaSection();

    tallenna();

    cy.wait('@createAmmToteutusResponse').then(({ request }) => {
      cy.wrap(request.body).snapshot();
    });
  });

  it('should be able to create korkeakoulu toteutus', () => {
    cy.route({
      method: 'GET',
      url: `**/koulutus/${koulutusOid}`,
      response: merge(koulutus({ tyyppi: 'yo' }), testKoulutusFields),
    });

    cy.route({
      method: 'PUT',
      url: '**/toteutus',
      response: {
        oid: '1.2.3.4.5.6',
      },
    }).as('createYoToteutusResponse');

    fillPohjaSection();
    fillKieliversiotSection();
    fillTiedotSection();

    cy.getByTestId('alempiOsaamisalatSection').within(() => {
      fillKkOsaamisalat();
      jatka();
    });

    cy.getByTestId('ylempiOsaamisalatSection').within(() => {
      fillKkOsaamisalat();
      jatka();
    });

    cy.getByTestId('jarjestamistiedotSection').within(() => {
      fillCommonJarjestamistiedot({ maksullisuusTyyppi: 'lukuvuosimaksu' });
      fillStipendi();
      jatka();
    });

    fillNayttamistiedotSection();
    fillJarjestajatSection();
    fillYhteystiedotSection();
    fillTilaSection();

    tallenna();

    cy.wait('@createYoToteutusResponse').then(({ request }) => {
      cy.wrap(request.body).snapshot();
    });
  });

  it('should be able to create lukio toteutus', () => {
    cy.route({
      method: 'GET',
      url: `**/koulutus/${koulutusOid}`,
      response: merge(koulutus({ tyyppi: 'lk' }), testKoulutusFields),
    });

    cy.route({
      method: 'PUT',
      url: '**/toteutus',
      response: {
        oid: '1.2.3.4.5.6',
      },
    }).as('createLkToteutusResponse');

    fillPohjaSection();
    fillKieliversiotSection();
    fillLukiolinjatSection();

    cy.getByTestId('jarjestamistiedotSection').within(() => {
      fillCommonJarjestamistiedot();
      fillStipendi();
      fillDiplomi();
      fillKielivalikoima();
      jatka();
    });

    fillNayttamistiedotSection();
    fillJarjestajatSection();
    fillYhteystiedotSection();
    fillTilaSection();

    tallenna();

    cy.wait('@createLkToteutusResponse').then(({ request }) => {
      cy.wrap(request.body).snapshot();
    });
  });
});
