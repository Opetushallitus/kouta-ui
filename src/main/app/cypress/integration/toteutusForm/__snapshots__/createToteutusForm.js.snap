exports[`createToteutusForm > should be able to create ammatillinen toteutus #0`] =
{
  "kielivalinta": [
    "fi"
  ],
  "koulutusOid": "1.2.1.1.1.1",
  "metadata": {
    "alemmanKorkeakoulututkinnonOsaamisalat": [],
    "aloituspaikat": null,
    "ammattinimikkeet": [
      {
        "arvo": "ammattinimike",
        "kieli": "fi"
      }
    ],
    "asiasanat": [
      {
        "arvo": "avainsana",
        "kieli": "fi"
      }
    ],
    "ilmoittautumislinkki": {},
    "kuvaus": {
      "fi": "Toteutuksen kuvaus"
    },
    "laajuus": null,
    "laajuusyksikkoKoodiUri": null,
    "lukiolinjaKoodiUri": null,
    "opetus": {
      "A1JaA2Kielivalikoima": [],
      "B1Kielivalikoima": [],
      "B2Kielivalikoima": [],
      "B3Kielivalikoima": [],
      "aidinkieliKielivalikoima": [],
      "diplomiKoodiUrit": [],
      "diplomiKuvaus": {},
      "koulutuksenAlkamiskausi": "kausi_0#1",
      "koulutuksenAlkamispaivamaara": null,
      "koulutuksenAlkamisvuosi": 2020,
      "koulutuksenPaattymispaivamaara": null,
      "koulutuksenTarkkaAlkamisaika": false,
      "lisatiedot": [
        {
          "otsikkoKoodiUri": "koulutuksenlisatiedot_0#1",
          "teksti": {
            "fi": "koulutuksenlisatiedot_0 kuvaus"
          }
        }
      ],
      "maksullisuusKuvaus": {
        "fi": "maksullisuus kuvaus"
      },
      "maksunMaara": 10,
      "muuKielivalikoima": [],
      "onkoMaksullinen": true,
      "onkoStipendia": false,
      "opetusaikaKoodiUrit": [
        "opetusaikakk_0#1"
      ],
      "opetusaikaKuvaus": {
        "fi": "opetusaika kuvaus"
      },
      "opetuskieletKuvaus": {
        "fi": "opetuskieli kuvaus"
      },
      "opetuskieliKoodiUrit": [
        "oppilaitoksenopetuskieli_0#1"
      ],
      "opetustapaKoodiUrit": [
        "opetuspaikkakk_0#1"
      ],
      "opetustapaKuvaus": {
        "fi": "opetustapa kuvaus"
      },
      "stipendinKuvaus": {},
      "stipendinMaara": null
    },
    "osaamisalat": [
      {
        "koodiUri": "osaamisala_0",
        "linkki": {
          "fi": "osaamisala_0 linkki"
        },
        "otsikko": {
          "fi": "osaamisala_0 otsikko"
        }
      }
    ],
    "suunniteltuKesto": {},
    "toteutusjaksot": [],
    "tutkinnonOsat": [],
    "tyyppi": "amm",
    "yhteyshenkilot": [
      {
        "nimi": {
          "fi": "nimi"
        },
        "puhelinnumero": {
          "fi": "puhelin"
        },
        "sahkoposti": {
          "fi": "sähkoposti"
        },
        "titteli": {
          "fi": "titteli"
        },
        "wwwSivu": {
          "fi": "verkkosivu"
        }
      }
    ],
    "ylemmanKorkeakoulututkinnonOsaamisalat": []
  },
  "muokkaaja": "1.2.246.562.24.62301161440",
  "nimi": {
    "fi": "toteutuksen nimi"
  },
  "organisaatioOid": "1.1.1.1.1.1",
  "tarjoajat": [
    "1.2.2.1.1.1"
  ],
  "teemakuva": null,
  "tila": "julkaistu"
};

exports[`createToteutusForm > should be able to create korkeakoulu toteutus #0`] =
{
  "kielivalinta": [
    "fi"
  ],
  "koulutusOid": "1.2.1.1.1.1",
  "metadata": {
    "alemmanKorkeakoulututkinnonOsaamisalat": [
      {
        "kuvaus": {
          "fi": "osaamisalan kuvaus"
        },
        "linkki": {
          "fi": "linkki"
        },
        "nimi": {
          "fi": "osaamisalan nimi"
        },
        "otsikko": {
          "fi": "osaamisalan otsikko"
        }
      }
    ],
    "aloituspaikat": null,
    "ammattinimikkeet": [
      {
        "arvo": "ammattinimike",
        "kieli": "fi"
      }
    ],
    "asiasanat": [
      {
        "arvo": "avainsana",
        "kieli": "fi"
      }
    ],
    "ilmoittautumislinkki": {},
    "kuvaus": {
      "fi": "Toteutuksen kuvaus"
    },
    "laajuus": null,
    "laajuusyksikkoKoodiUri": null,
    "lukiolinjaKoodiUri": null,
    "opetus": {
      "A1JaA2Kielivalikoima": [],
      "B1Kielivalikoima": [],
      "B2Kielivalikoima": [],
      "B3Kielivalikoima": [],
      "aidinkieliKielivalikoima": [],
      "diplomiKoodiUrit": [],
      "diplomiKuvaus": {},
      "koulutuksenAlkamiskausi": "kausi_0#1",
      "koulutuksenAlkamispaivamaara": null,
      "koulutuksenAlkamisvuosi": 2020,
      "koulutuksenPaattymispaivamaara": null,
      "koulutuksenTarkkaAlkamisaika": false,
      "lisatiedot": [
        {
          "otsikkoKoodiUri": "koulutuksenlisatiedot_0#1",
          "teksti": {
            "fi": "koulutuksenlisatiedot_0 kuvaus"
          }
        }
      ],
      "maksullisuusKuvaus": {
        "fi": "maksullisuus kuvaus"
      },
      "maksunMaara": null,
      "muuKielivalikoima": [],
      "onkoMaksullinen": false,
      "onkoStipendia": true,
      "opetusaikaKoodiUrit": [
        "opetusaikakk_0#1"
      ],
      "opetusaikaKuvaus": {
        "fi": "opetusaika kuvaus"
      },
      "opetuskieletKuvaus": {
        "fi": "opetuskieli kuvaus"
      },
      "opetuskieliKoodiUrit": [
        "oppilaitoksenopetuskieli_0#1"
      ],
      "opetustapaKoodiUrit": [
        "opetuspaikkakk_0#1"
      ],
      "opetustapaKuvaus": {
        "fi": "opetustapa kuvaus"
      },
      "stipendinKuvaus": {
        "fi": "stipendi kuvaus"
      },
      "stipendinMaara": 20
    },
    "osaamisalat": [],
    "suunniteltuKesto": {},
    "toteutusjaksot": [],
    "tutkinnonOsat": [],
    "tyyppi": "yo",
    "yhteyshenkilot": [
      {
        "nimi": {
          "fi": "nimi"
        },
        "puhelinnumero": {
          "fi": "puhelin"
        },
        "sahkoposti": {
          "fi": "sähkoposti"
        },
        "titteli": {
          "fi": "titteli"
        },
        "wwwSivu": {
          "fi": "verkkosivu"
        }
      }
    ],
    "ylemmanKorkeakoulututkinnonOsaamisalat": [
      {
        "kuvaus": {
          "fi": "osaamisalan kuvaus"
        },
        "linkki": {
          "fi": "linkki"
        },
        "nimi": {
          "fi": "osaamisalan nimi"
        },
        "otsikko": {
          "fi": "osaamisalan otsikko"
        }
      }
    ]
  },
  "muokkaaja": "1.2.246.562.24.62301161440",
  "nimi": {
    "fi": "toteutuksen nimi"
  },
  "organisaatioOid": "1.1.1.1.1.1",
  "tarjoajat": [
    "1.2.2.1.1.1"
  ],
  "teemakuva": null,
  "tila": "julkaistu"
};

exports[`createToteutusForm > should be able to create lukio toteutus #0`] =
{
  "kielivalinta": [
    "fi"
  ],
  "koulutusOid": "1.2.1.1.1.1",
  "metadata": {
    "alemmanKorkeakoulututkinnonOsaamisalat": [],
    "aloituspaikat": null,
    "ammattinimikkeet": [
      {
        "arvo": "ammattinimike",
        "kieli": "fi"
      }
    ],
    "asiasanat": [
      {
        "arvo": "avainsana",
        "kieli": "fi"
      }
    ],
    "ilmoittautumislinkki": {},
    "kuvaus": {},
    "laajuus": null,
    "laajuusyksikkoKoodiUri": null,
    "lukiolinjaKoodiUri": "lukiolinjat_0#1",
    "opetus": {
      "A1JaA2Kielivalikoima": [
        "kieli_0#1"
      ],
      "B1Kielivalikoima": [
        "kieli_3#1"
      ],
      "B2Kielivalikoima": [
        "kieli_1#1"
      ],
      "B3Kielivalikoima": [
        "kieli_4#1"
      ],
      "aidinkieliKielivalikoima": [
        "kieli_2#1"
      ],
      "diplomiKoodiUrit": [
        "lukiodiplomit_0#1"
      ],
      "diplomiKuvaus": {
        "fi": "Diplomi kuvaus"
      },
      "koulutuksenAlkamiskausi": "kausi_0#1",
      "koulutuksenAlkamispaivamaara": null,
      "koulutuksenAlkamisvuosi": 2020,
      "koulutuksenPaattymispaivamaara": null,
      "koulutuksenTarkkaAlkamisaika": false,
      "lisatiedot": [
        {
          "otsikkoKoodiUri": "koulutuksenlisatiedot_0#1",
          "teksti": {
            "fi": "koulutuksenlisatiedot_0 kuvaus"
          }
        }
      ],
      "maksullisuusKuvaus": {
        "fi": "maksullisuus kuvaus"
      },
      "maksunMaara": 10,
      "muuKielivalikoima": [
        "kieli_4#1"
      ],
      "onkoMaksullinen": true,
      "onkoStipendia": true,
      "opetusaikaKoodiUrit": [
        "opetusaikakk_0#1"
      ],
      "opetusaikaKuvaus": {
        "fi": "opetusaika kuvaus"
      },
      "opetuskieletKuvaus": {
        "fi": "opetuskieli kuvaus"
      },
      "opetuskieliKoodiUrit": [
        "oppilaitoksenopetuskieli_0#1"
      ],
      "opetustapaKoodiUrit": [
        "opetuspaikkakk_0#1"
      ],
      "opetustapaKuvaus": {
        "fi": "opetustapa kuvaus"
      },
      "stipendinKuvaus": {
        "fi": "stipendi kuvaus"
      },
      "stipendinMaara": 20
    },
    "osaamisalat": [],
    "suunniteltuKesto": {},
    "toteutusjaksot": [],
    "tutkinnonOsat": [],
    "tyyppi": "lk",
    "yhteyshenkilot": [
      {
        "nimi": {
          "fi": "nimi"
        },
        "puhelinnumero": {
          "fi": "puhelin"
        },
        "sahkoposti": {
          "fi": "sähkoposti"
        },
        "titteli": {
          "fi": "titteli"
        },
        "wwwSivu": {
          "fi": "verkkosivu"
        }
      }
    ],
    "ylemmanKorkeakoulututkinnonOsaamisalat": []
  },
  "muokkaaja": "1.2.246.562.24.62301161440",
  "nimi": {
    "fi": "lukiolinjat_0"
  },
  "organisaatioOid": "1.1.1.1.1.1",
  "tarjoajat": [
    "1.2.2.1.1.1"
  ],
  "teemakuva": null,
  "tila": "julkaistu"
};
